dask/operation.py

```diff
@@ -0,0 +1,384 @@
+from __future__ import annotations
+
+import copy
+from collections import defaultdict
+from collections.abc import Mapping
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Callable
+
+from dask.base import tokenize
+from dask.core import reverse_dict
+from dask.highlevelgraph import HighLevelGraph
+from dask.optimization import SubgraphCallable
+
+
+@dataclass(frozen=True)
```

Comment:
As long as this and `FusableOperation` don't have any attributes, I don't think they need to be dataclasses at all, they could just be abstract interfaces/protocols

Reply:
Good point!